projects/scion/workbench/src/lib/view/workbench-view.registry.spec.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2018-2023 Swiss Federal Railways
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+import {TestBed} from '@angular/core/testing';
+import {WorkbenchViewRegistry} from './workbench-view.registry';
+import {ObserveCaptor} from '@scion/toolkit/testing';
+import {expect} from '../testing/jasmine/matcher/custom-matchers.definition';
+import {WorkbenchView} from '../view/workbench-view.model';
+import {WorkbenchLayoutService} from '../layout/workbench-layout.service';
+import {ViewComponent} from './view.component';
+import {ɵWorkbenchView} from './ɵworkbench-view.model';
+import {WORKBENCH_LAYOUT_INITIAL_PART_ID, WorkbenchLayoutFactory} from '../layout/workbench-layout-factory.service';
+import {WorkbenchUrlObserver} from '../routing/workbench-url-observer.service';
+import {WorkbenchTestingModule} from '../testing/workbench-testing.module';
+import {WorkbenchRouter} from '../routing/workbench-router.service';
+import {RouterTestingModule} from '@angular/router/testing';
+import {TestComponent} from '../testing/test.component';
+import {mapArray} from '@scion/toolkit/operators';
+
+describe('WorkbenchViewRegistry', () => {
+
+  it('should delay emitting views until the next layout change', async () => {
+    TestBed.configureTestingModule({
+      imports: [
+        WorkbenchTestingModule.forTest(),
+      ],
+      providers: [
+        {provide: WorkbenchUrlObserver, useValue: null}, // disable WorkbenchUrlObserver
+      ],
+    });
+
+    // Register view 'view.1'.
+    TestBed.inject(WorkbenchViewRegistry).register(TestBed.runInInjectionContext(() => new ɵWorkbenchView('view.1', {component: ViewComponent})));
+
+    // Expect registry to emit registered views upon subscription.
+    const viewsCaptor = new ObserveCaptor();
+    TestBed.inject(WorkbenchViewRegistry).views$.subscribe(viewsCaptor);
+    expect(viewsCaptor.getValues()).toEqual([
```

Comment:
There is no layout change yet, is this really anticipated?

Reply:
Yes, it's the initial emission upon subscription.